src/GraphQL.EntityFramework/Where/ExpressionBuilder.cs

```diff
@@ -1,109 +1,342 @@
+                        break;
                 }
             }
-        }
-
-        internal static Expression<Func<T, bool>> BuildSinglePredicate(string path, Comparison comparison, string value, StringComparison? stringComparison = null)
-        {
-            var property = PropertyCache<T>.GetProperty(path);
-
-            if (property.PropertyType == typeof(string))
-            {
-                WhereValidator.ValidateSingleString(comparison, stringComparison);
-                return BuildStringCompare(comparison, value, property, stringComparison);
-            }
 
-            WhereValidator.ValidateSingleObject(property.PropertyType, comparison, stringComparison);
-            return BuildObjectCompare(comparison, value, property);
+            return expressionBody;
         }
 
-        static Expression<Func<T, bool>> BuildStringCompare(Comparison comparison, string? value, Property<T> property, StringComparison? stringComparison)
-        {
-            var body = MakeStringComparison(property.Left, comparison, value, stringComparison);
-            return Expression.Lambda<Func<T, bool>>(body, property.SourceParameter);
-        }
+        #endregion
 
-        static Expression<Func<T, bool>> BuildObjectCompare(Comparison comparison, string? value, Property<T> property)
-        {
-            var valueObject = TypeConverter.ConvertStringToType(value, property.PropertyType);
-            var body = MakeObjectComparison(property.Left, comparison, valueObject);
-            return Expression.Lambda<Func<T, bool>>(body, property.SourceParameter);
-        }
 
-        static Expression<Func<T, bool>> BuildObjectIn(string[] values, Property<T> property, bool not = false)
+        #region Operations
+
+        /// <summary>
+        /// Make Object List In Comparision
+        /// </summary>
+        /// <param name="values"></param>
+        /// <param name="property"></param>
+        /// <returns></returns>
+        static Expression MakeObjectIn(string[] values, Property<T> property)
         {
+            // Attempt to convert the string values to the object type
             var objects = TypeConverter.ConvertStringsToList(values, property.PropertyType);
+            // Make the object values a constant expression
             var constant = Expression.Constant(objects);
-            var body = Expression.Call(constant, property.ListContainsMethod, property.Left);
-            return Expression.Lambda<Func<T, bool>>(not ? Expression.Not(body) : (Expression) body, property.SourceParameter);
+            // Build and return the expression body
+            return Expression.Call(constant, property.ListContainsMethod, property.Left);
         }
 
-        static Expression<Func<T, bool>> BuildStringIn(string[] array, Property<T> property, StringComparison? comparison, bool not = false)
+        /// <summary>
+        /// Make String List In Comparison
+        /// </summary>
+        /// <param name="values"></param>
```

Comment:
can u remove the empty params and returns

Reply:
yep